andela/wg-aces
# #159250137 Create users through api

#### What does this PR do?
It enables creation of users through the api.

#### Description of Task to be completed?
Users can now request for access to be able to create users through the api. This right is granted to them by the admin on demand. After which they can start to create users through the api.

#### How should this be manually tested?;
- Clone the repository 
- Run migrations `python manage.py migrate`
- Login into the application as a user.
- Select  `About this software` tab and select `rest api` from the resulting dropdown and click button request for access to create users through api
- Generate an api key for that user.
- Go the command line and enter `python manage.py users-requesting-access` to get the username of users requesting access.
- To grant permission to the specific users.                                           
- Enter `python manage.py add-user-rest-api --username USERNAME`. Replace USERNAME with actual username of user you want to grant permission to.
- You can later enter `python manage.py list-user-rest-api` to see the users created through the api
- Now you can go and test out this endpoint `/api/v2/user/` to create users through the api.
- You should pass the api key generated and the user should have been granted permission to create users through api. 
- The json data fields required for creation of user are `username, password, email`.

#### What are the relevant pivotal tracker stories?
[#159250137 ](https://www.pivotaltracker.com/story/show/159250137)

#### Screenshots
<img width="916" alt="screen shot 2018-08-16 at 19 56 51" src="https://user-images.githubusercontent.com/39958516/44222918-c5289800-a18e-11e8-886b-3aa867a055e7.png">
